### ISSUE TITLE
Failure to Create Server

### ISSUE THREAD

**Issue by DeathRabbit679** [Solvik/netbox-agent#230]
**Describe the bug**
When attempting to create a new server into my netbox instance, netbox-agent bombs with the following trace

```
Traceback (most recent call last):
  File "/home/foouser/netbox-agent/venv/bin/netbox_agent", line 8, in <module>
    sys.exit(main())
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/netbox_agent/cli.py", line 50, in main
    return run(config)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/netbox_agent/cli.py", line 43, in run
    server.netbox_create_or_update(config)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/netbox_agent/server.py", line 407, in netbox_create_or_update
    server = self._netbox_create_server(datacenter, tenant, rack)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/netbox_agent/server.py", line 281, in _netbox_create_server
    tags=[{'name': x} for x in self.tags],
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/pynetbox/core/endpoint.py", line 309, in create
    ).post(args[0] if args else kwargs)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/pynetbox/core/query.py", line 387, in post
    return self._make_call(verb="post", data=data)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/pynetbox/core/query.py", line 266, in _make_call
    url_override or self.url, headers=headers, params=params, json=data
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/requests/sessions.py", line 577, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/requests/sessions.py", line 515, in request
    prep = self.prepare_request(req)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/requests/sessions.py", line 453, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/requests/models.py", line 321, in prepare
    self.prepare_body(data, files, json)
  File "/home/foouser/netbox-agent/venv/lib/python3.6/site-packages/requests/models.py", line 473, in prepare_body
    body = complexjson.dumps(json, allow_nan=False)
  File "/usr/lib64/python3.6/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib64/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib64/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'Record' is not JSON serializable
```
I believe this is because as of this commit https://github.com/Solvik/netbox-agent/commit/dfb6b234ba3c63db69b4e4100aec28c6efe551c4 , we are no longer getting the id and passing along the whole `Record` which pynetbox just passes on through to requests, causing the JSON error.

**Expected behavior**
Being able to add a server to netbox

**Configuration file**
```
netbox:
  url: 'http://4.3.2.1'
  token: tokentokentokentoken
  # uncomment to disable ssl verification
  # ssl_verify: false

network:
  ignore_interfaces: "(dummy.*|docker.*)"
  ignore_ips: (127\.0\.0\..*)
  # enable auto-cabling
  lldp: false

datacenter_location:
  driver: "file:/root/datacenter"
  regex: "(.*)"


inventory: true
```

**Environment:**
 - OS: Cent7
 - Netbox agent version v0.7.2

**Additional context**
Changing https://github.com/Solvik/netbox-agent/blob/be770a947a85d29c832a063d08a23bc484f15dbc/netbox_agent/server.py#L277 to use `self.device_platform.id` fixes it for me.

**Comment by KivraChristoffer** [comment]
Could this be closed ?

**Comment by DeathRabbit679** [comment]
Fixed by #231
